src/shared.ts

```diff
@@ -96,6 +113,10 @@ export const loadScript = (
         console.warn(EXISTING_SCRIPT_MESSAGE);
       } else if (!script) {
         script = injectScript(params);
+      } else {
+        // if script exists, but we are reloading due to an error,
+        // reload script to trigger 'load' event
+        script = reloadScript(params);
```

Comment:
Instead of querying the DOM for the script again, it would be simpler to remove it from the DOM and then reassign it. Here's my suggestion.
```suggestion
      } else {
        // If `stripePromise` is unresolved but a script exists, there was an error loading it.
        // We remove and reload the script anew.
        script.parentNode.removeChild(script);
        script = injectScript(params);
```